flight/search/plot_algo.py

```diff
@@ -0,0 +1,601 @@
+        a_star: AStarFinder
+            the A* searcher object, used as class variable to avoid overhead of
+            repeated initializations
+        a_star_grid: Grid
+            the A* search grid, used as class variable to avoid overhead of
+            repeated initializations
+        move_list: list[tuple[int, int]]
+            the list of valid moves in the grid ((1, 1) represents moving diagonally,
+            for example)
+
+    Methods
+    -------
+    get_num_valids() -> int
+        returns number of non-empty cells
+    get_valid_moves(history: list[tuple[int, int]]) -> list[tuple[int, int]]
+        returns all possible moves based on the current position
+    valid_solution(solution: list[tuple[int, int]]) -> bool
+        determines if a given path touches all cells
+    find_unseens(history: list[tuple[int, int]]) -> list[tuple[int, int]]
+        finds all cells not visited in the given path
+    find_closest(points: list[tuple[int, int]], px: tuple[int, int]) -> tuple[int, int]
+        finds the closest point to px
+    in_corner(pos: tuple[int, int], history[list[tuple[int, int]]])
+        returns the path to escape the given corner
+    breadth_search(start: tuple[int, int]) -> list[tuple[int, int]]
+        returns the shrotest circuit route through all cells.
+    """
+
+    def get_num_valids(self) -> int:
+        """
+        Retuns the number of valid compressed cells
+
+        Returns
+        -------
+        num_valids: int
+            The number of valid compressed cells
+        """
+        num: int = 0
+        i: int
+        j: int
+        for i, _ in enumerate(self.compressed):
+            for j, _ in enumerate(self.compressed[0]):
+                if self.compressed[i][j] != 0:
+                    num += 1
+        return num
+
+    def __init__(self, cell_map: CellMap, view_radius: int) -> None:
+        """
+        Initializes the object
+
+        Parameters
+        ----------
+        cell_map: CellMap
+            the cell map being searched
+        view_radius: int
+            how many cells away the searcher can see
+        """
+        self.compressed: NDArray[int8] = Compressor.compress(view_radius, cell_map)
+        self.num_valids: int = self.get_num_valids()
+        self.a_star: AStarFinder = AStarFinder()
+        self.a_star_grid: Grid = Grid(matrix=self.compressed)
+        self.move_list: list[tuple[int, int]] = [
+            (1, 0),
+            (-1, 0),
+            (0, 1),
+            (0, -1),
+            (1, 1),
+            (-1, -1),
+        ]
+
+    def get_valid_moves(self, history: list[tuple[int, int]]) -> list[tuple[int, int]]:
+        """
+        Given a list of moves already made on the compressed map,
+        return all possible immediate moves.
+
+        Parameters
+        ----------
+        history : list[tuple[int, int]]
+            list of i, j positions visited thus far
```

Comment:
Change i,j position comment here to indicate row,column - just so it makes the comment slightly more readable.